openscenario/map_fragment/include/map_fragment/geometry.hpp

```diff
@@ -0,0 +1,306 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MAP_FRAGMENT__GEOMETRY__HPP_
+#define MAP_FRAGMENT__GEOMETRY__HPP_
+
+#include <cmath>
+#include <string>
+#include <algorithm>
+#include <memory>
+#include <stdexcept>
+
+namespace map_fragment
+{
+
+struct Point2d
+{
+  double x = 0;
+  double y = 0;
+};
+
+struct Vector2d
+{
+  double x = 0;
+  double y = 0;
+};
+
+auto operator+(Point2d const& a, Vector2d const& b) -> Point2d
+{
+  return {a.x + b.x, a.y + b.y};
+}
+
+auto operator*(Vector2d const& v, double multiplier) -> Vector2d
+{
+  return {v.x * multiplier, v.y * multiplier};
+}
+
+auto operator-(Point2d const& a, Point2d const& b) -> Vector2d
+{
+  return {a.x - b.x, a.y - b.y};
+}
+
+auto operator+(Vector2d const& a, Vector2d const& b) -> Vector2d
+{
+  return {a.x + b.x, a.y + b.y};
+}
+
+auto rotate(Vector2d const& v, double angle) -> Vector2d
+{
+  return {
+    v.x * std::cos(angle) - v.y * std::sin(angle),
+    v.x * std::sin(angle) + v.y * std::cos(angle)
+  };
+}
+
+auto rotate(Point2d const& p, double angle) -> Point2d
+{
+  Point2d p0 = {0, 0};
+  return p0 + rotate(p - p0, angle);
+}
+
+auto calculateAngle(Vector2d const& v) -> double
+{
+  return std::atan2(v.y, v.x);
+}
+
+struct Transformation2d
+{
+  Vector2d translation;
+  double rotation;
+};  // struct Transformation2d
+
+auto applyTransformation(Point2d const& p, Transformation2d const& t) -> Point2d
+{
+  return rotate(p, t.rotation) + t.translation;
+}
+
+auto applyTransformation(Vector2d const& v, Transformation2d const& t) -> Vector2d
+{
+  return rotate(v, t.rotation);
+}
+
+auto chainTransformations(Transformation2d const& first,
+                          Transformation2d const& second) -> Transformation2d
+{
+  Transformation2d t;
+  t.translation = first.translation + rotate(second.translation, first.rotation);
+  t.rotation = first.rotation + second.rotation;
+  return t;
+}
+
+/**
+ * Parametric representation of a smooth curve in 2D Cartesian space
+ * 
+ * By convention, the curve is assumed to always start at point (0, 0)
+ * and have a tangent vector there of [1, 0]. Geometric trasformations
+ * can be applied afterwards if other start positions and/or
+ * directions are required.
+ */
+class ParametricCurve
+{
+
+public:
+  using Pointer = std::shared_ptr<ParametricCurve>;
+  
+  /**
+   * Calculate curve point for given parameter t ∈ [0, 1]
+   */
+  auto getPosition(double t) -> Point2d
```

Comment:
Please change the `t` in the variable and formal argument names in this pull request to descriptive names. In the `scenario_simulator_v2` repository, `t` can be interpreted in various ways. For example, time `t` or `t` in the Frenet coordinate system. Use `time` in the former case and `tangent` in the latter case so that the reader does not have to guess the meaning of `t`.

At least in this file, `t` appears to have multiple meanings, including `t`, which means transform.

Reply:
Replaced `t` with `tangent` when using the variable as a Frenet coordinate.

> [...] including `t`, which means transform

Do you mean in that case I should keep using `t` (and not for example `transform`)?

Reply:
> Do you mean in that case I should keep using t (and not for example transform)?

Sorry for the confusing explanation.
What I meant was "I want all `t`s in this pull request to be renamed to descriptive names."
Please also change the name `t`, which seems to mean transform, to something more descriptive.

Reply:
[Done during refactor](https://github.com/tier4/scenario_simulator_v2/pull/1108/commits/84f296ce44357f764116838a99f9c2a46d087b2f). New functions based on Eigen-provided types do not use ambiguous parameter names `t`.